ClearKeep/Scenes/Home/Models/CallViewModel.swift

```diff
@@ -0,0 +1,454 @@
+//
+//  CallViewModel.swift
+//  ClearKeep
+//
+//  Created by HOANDHTB on 19/07/2022.
+//
+
+import Foundation
+import WebRTC
+import ChatSecure
+import ChatSecure
+
+class CallViewModel: NSObject, ObservableObject {
+	@Published var localVideoView: RTCMTLEAGLVideoView?
+	@Published var remoteVideoView: RTCMTLEAGLVideoView?
+	@Published var remotesVideoView = [RTCMTLEAGLVideoView]()
+	@Published var cameraOn = true
+	@Published var cameraFront = false
+	@Published var microEnable = true
+	@Published var speakerEnable = false
+	@Published var callStatus: CallStatus = .calling
+	@Published var callGroup = false
+	@Published var timeCall = ""
+	@Published var remoteViewRenderSize: CGSize = CGSize.zero
+	@Published var isVideoRequesting = false
+	@Published var callType: CallType = .audio
+	@Published var remotesVideoViewConfig = [String : CustomVideoViewConfig]()
+	var remotesVideoViewDict = [String : RTCMTLEAGLVideoView]()
+	var backHandler: (() -> Void)? = nil
+	
+	enum RenderScaleMode {
+		case scaleToFit
+		case scaleToFill
+	}
+	
+	public var callBox: CallBox?
+	var callTimer: Timer?
+	lazy var timeCounter = TimeCounter()
+	
+	var timeoutTimer: Timer?
+	var callInterval: Int = 0
+	
+	override init() {
+		super.init()
+	}
+	
+	func updateCallBox(callBox: CallBox) {
+		self.callBox = callBox
+		self.callType = callBox.type
+		
+		if callBox.type == .audio {
+			self.cameraOn = false
+			self.speakerEnable = false
+		} else {
+			self.cameraOn = callBox.isCamera
+			self.speakerEnable = true
+		}
+		
+		updateMicroConfig()
+		updateSpeakerConfig()
+		updateCameraConfig()
+		
+		updateVideoView()
+
+		self.callBox?.stateDidChange = { [weak self] in
+			DispatchQueue.main.async {
+				let boxStatus = self?.callBox?.status ?? .ended
+				if boxStatus == .answered {
+					self?.startCallTimer()
+					if let type = self?.callBox?.type, type == .video {
+						self?.forceEnableSpeaker()
+					}
+				} else if self?.callStatus != .ringing, boxStatus == .ringing {
+					self?.startCallTimout()
+				} else if boxStatus == .ended {
+					self?.stopCallTimer()
+					self?.stopTimeoutTimer()
+				}
+				self?.callStatus = self?.callBox?.status ?? .ended
+				self?.updateVideoView()
+			}
+		}
+		
+		self.callBox?.renderSizeChangeWithSize = { [weak self] (size, uId) in
+			guard let self = self else {
+				return
+			}
+			print("uId", uId)
+			// Just support only call 1:1 for now
+			self.remoteViewRenderSize = size
+		}
+		
+		self.callBox?.membersInCallDidChange = { [weak self] in
+			guard let self = self else {
+				return
+			}
+			self.updateVideoView()
+		}
+	}
+	
+	func getStatusMessage() -> String {
+		switch callStatus {
+		case .calling:
+			return "Calling..."
+		case .ringing:
+			return "Connecting..."
+		case .ended:
+			return "End call..."
+		default:
+			return ""
+		}
+	}
+	
+	func getUserName() -> String {
+		return callBox?.clientName ?? "N/A"
+	}
+	
+	func getAvatar() -> String {
+		return callBox?.avatar ?? "" //""
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/fe9f474e9dacdb67166a3b18715e917932515a0f/ClearKeep/Scenes/Home/Models/CallViewModel.swift#L119"></span>Prefer at least one space after slashes for comments.<br />
<code>comment_spacing</code> <code>CallViewModel.swift:119</code></td>
    </tr>
  </tbody>
</table>